libraries/opensk/src/ctap/hid/mod.rs

```diff
@@ -306,6 +334,26 @@ impl<E: Env> CtapHid<E> {
                 None
             }
             CtapHidCommand::Wink => Some(message),
+            CtapHidCommand::Lock => {
+                if message.payload.len() != 1 {
+                    return Some(Self::error_message(cid, CtapHidError::InvalidLen));
+                }
+                if message.payload[0] > 10 {
+                    return Some(Self::error_message(cid, CtapHidError::InvalidPar));
+                }
+                if message.payload[0] == 0 {
```

Comment:
Hard to make sure of the whole flow but I assume that `cid` will be compared before calling this `preprocess_message()` method otherwise any channel could remove the lock with a `0` parameter.

It would be nice to add a test to check that if `cid` lock, `cid+1` doesn't unlock